core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c

```diff
@@ -495,6 +495,12 @@ emscripten_notify_memory_growth_wrapper(wasm_exec_env_t exec_env, int i)
     (void)i;
 }
 
+static void
+emscripten_sleep_wrapper(wasm_exec_env_t exec_env, int timeout_ms)
+{
+    usleep(timeout_ms * 1000);
```

Comment:
integer overflow may occur in `timeout_ms * 1000`, had better check it?
```C
    if (timeout > UINT32_MAX /1000)
        usleep(UINT32_MAX);
    else
        usleep(timeout_ms * 1000);
```

Reply:
Thanks, your comments made me realize that the implementation needs to be more robust in several ways, so I've rewritten it.

Reply:
Sorry about that, it's pushed now.